File: C/Algorithms/2427.c
int commonFactors(int a, int b){
    if (a == 1 || b == 1)
        return 1;
    int fractions = 1;
    int max, min;
    if (a > b)
        max = a, min = b;
    else
        max = b, min = a;
    if (max % min == 0)
        fractions++;
    
    for(int i = 2; i < (min / 2) + 1; i++){
        if (max % i == 0 && min % i == 0)
            fractions++;
    }
    return fractions;
}

File: Rust/Algorithms/2427.rs
impl Solution {
    pub fn common_factors(a: i32, b: i32) -> i32 {
        if a == 1 || b == 1 {
            return 1;
        }
        let mut fractions = 1;
        let (max, min) = if a > b { (a, b) } else { (b, a) };
        if max % min == 0 {
            fractions += 1;
        }

        for i in 2..((min / 2) + 1) {
            if max % i == 0 && min % i == 0 {
                fractions += 1;
            }
        }
        fractions
    }
}